gateway/modules/crud/operations.go

```diff
@@ -160,6 +160,32 @@ func (m *Module) Delete(ctx context.Context, dbAlias, project, col string, req *
 	return err
 }
 
+// ExecPreparedQuery removes the documents(s) which match a query from the database based on dbType
+func (m *Module) ExecPreparedQuery(ctx context.Context, project, dbAlias, id string, req *model.PreparedQueryRequest) error {
+	m.RLock()
+	defer m.RUnlock()
+
+	crud, err := m.getCrudBlock(dbAlias)
+	if err != nil {
+		return err
+	}
+
+	if err := crud.IsClientSafe(); err != nil {
+		return err
+	}
+
+	var args []interface{}
+	for i := 0; i < len(m.queries.Arguments); i++ {
```

Comment:
m.queries will be a map. You'll have to first check if the provided id exists in the map. Throw error if it doesn't